app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_data = dict()
+
+    def wrapper(*args: Any) -> int | str:
```

Comment:
If you're trying to make a general caching decorator, using Any as the return type would be better